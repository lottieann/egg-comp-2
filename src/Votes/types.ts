export interface VotesData {
  name: string,
  email: string,
  first: string,
  second: string,
  third: string,
  fourth: string,
  fifth: string
}
