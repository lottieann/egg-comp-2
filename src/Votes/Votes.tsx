import * as React from 'react'

import styled from 'styled-components'

const Container = styled.div`
  margin-top: 15vh;
  flex-direction: column;
  display: flex;
  justify-content: center;
  align-items: center;
`

export const Votes = () => {

  return (
    <Container>

      <iframe
        src="https://docs.google.com/forms/d/e/1FAIpQLSdJKFyzMuUCj0dnoYSt-QUIgMlDnIWDuTno4cyZqPCjAme_kg/viewform?embedded=true"
        width="320"
        height="1515"
        frameBorder="0"
      >
        Loadingâ€¦
      </iframe>
    </Container>
  )
}
