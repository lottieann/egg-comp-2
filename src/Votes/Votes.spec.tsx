import * as React from 'react'

import 'jest-styled-components'

import '@testing-library/jest-dom/extend-expect'
import { fireEvent, render, waitFor } from '@testing-library/react'
import { Votes } from './Votes'

const setup = () => {
  return <Votes />
}

describe('Votes', () => {
  it('should throw an alert if form name field is empty', async () => {
    const { getByPlaceholderText, getByRole, getByTestId } = render(setup())
    const jsdomAlert = window.alert // remember the jsdom alert
    window.alert = () => {}
    fireEvent.change(getByPlaceholderText('Enter an email'), {
      target: {
        value: 'hello@live.com',
      },
    })

    fireEvent.click(getByTestId('submit-button'))

    expect(getByRole('alert')).toHaveTextContent('Please enter your name')
    window.alert = jsdomAlert
  })
  it('should call submit form when button clicked', () => {
    const { getByTestId } = render(setup())

    fireEvent.click(getByTestId('submit-button'))
  })
  it('should reload page when reload button is clicked', () => {
    // const { getByTestId } = render(setup())
    // fireEvent.click(getByTestId('reload-button'))
  })
  it('should render Votes correctly', () => {
    const { container } = render(setup())
    expect(container).toMatchSnapshot()
  })
})
