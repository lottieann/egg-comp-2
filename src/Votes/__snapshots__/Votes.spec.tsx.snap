// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Votes should render Votes correctly 1`] = `
.c1 {
  width: 200px;
}

.c2 {
  width: 150px;
}

.c0 {
  margin-left: 200px;
}

<div>
  <div
    class="c0"
  >
    <p>
      You get 
      <b>
        5
      </b>
       votes, use them wisely!
    </p>
    <form>
      <div
        class="c1"
      >
        <label>
          My favourite
        </label>
        <select
          class="c2"
        >
          <option
            value="1"
          >
            1
          </option>
          <option
            value="2"
          >
            2
          </option>
          <option
            value="3"
          >
            3
          </option>
          <option
            value="4"
          >
            4
          </option>
          <option
            value="5"
          >
            5
          </option>
          <option
            value="6"
          >
            6
          </option>
          <option
            value="7"
          >
            7
          </option>
          <option
            value="8"
          >
            8
          </option>
          <option
            value="9"
          >
            9
          </option>
        </select>
      </div>
      <div
        class="c1"
      >
        <label>
          I loved this
        </label>
        <select
          class="c2"
        >
          <option
            value="1"
          >
            1
          </option>
          <option
            value="2"
          >
            2
          </option>
          <option
            value="3"
          >
            3
          </option>
          <option
            value="4"
          >
            4
          </option>
          <option
            value="5"
          >
            5
          </option>
          <option
            value="6"
          >
            6
          </option>
          <option
            value="7"
          >
            7
          </option>
          <option
            value="8"
          >
            8
          </option>
          <option
            value="9"
          >
            9
          </option>
        </select>
      </div>
      <div
        class="c1"
      >
        <label>
          This too
        </label>
        <select
          class="c2"
        >
          <option
            value="1"
          >
            1
          </option>
          <option
            value="2"
          >
            2
          </option>
          <option
            value="3"
          >
            3
          </option>
          <option
            value="4"
          >
            4
          </option>
          <option
            value="5"
          >
            5
          </option>
          <option
            value="6"
          >
            6
          </option>
          <option
            value="7"
          >
            7
          </option>
          <option
            value="8"
          >
            8
          </option>
          <option
            value="9"
          >
            9
          </option>
        </select>
      </div>
      <div
        class="c1"
      >
        <label>
          I thought this was good
        </label>
        <select
          class="c2"
        >
          <option
            value="1"
          >
            1
          </option>
          <option
            value="2"
          >
            2
          </option>
          <option
            value="3"
          >
            3
          </option>
          <option
            value="4"
          >
            4
          </option>
          <option
            value="5"
          >
            5
          </option>
          <option
            value="6"
          >
            6
          </option>
          <option
            value="7"
          >
            7
          </option>
          <option
            value="8"
          >
            8
          </option>
          <option
            value="9"
          >
            9
          </option>
        </select>
      </div>
      <div
        class="c1"
      >
        <label>
          Okay, I guess...
        </label>
        <select
          class="c2"
        >
          <option
            value="1"
          >
            1
          </option>
          <option
            value="2"
          >
            2
          </option>
          <option
            value="3"
          >
            3
          </option>
          <option
            value="4"
          >
            4
          </option>
          <option
            value="5"
          >
            5
          </option>
          <option
            value="6"
          >
            6
          </option>
          <option
            value="7"
          >
            7
          </option>
          <option
            value="8"
          >
            8
          </option>
          <option
            value="9"
          >
            9
          </option>
        </select>
      </div>
      <input
        type="submit"
      />
    </form>
  </div>
</div>
`;
